agent/api/task/task.go

```diff
@@ -3617,42 +3617,42 @@ func (task *Task) IsLaunchTypeFargate() bool {
 //
 // * GPU
 //   - Concatenate each container's gpu ids
-func (task *Task) ToHostResources() map[string]*ecs.Resource {
-	resources := make(map[string]*ecs.Resource)
+func (task *Task) ToHostResources() map[string]ecstypes.Resource {
```

Comment:
to make all of these type conversions simpler, if memory, CPU, etc changed from int64/float64 to int32, can we update the Task object: 

https://github.com/aws/amazon-ecs-agent/blob/447b0eb52163a31483bd1104bc8239b1ad43f42f/agent/api/task/task.go#L192-L196